pythonforandroid/bootstraps/webview/build/src/main/java/org/kivy/android/PythonActivity.java

```diff
@@ -355,21 +355,27 @@ public static ViewGroup getLayout() {
     long lastBackClick = SystemClock.elapsedRealtime();
     @Override
     public boolean onKeyDown(int keyCode, KeyEvent event) {
-        // Check if the key event was the Back button and if there's history
-        if ((keyCode == KeyEvent.KEYCODE_BACK) && mWebView.canGoBack()) {
-            mWebView.goBack();
-            return true;
-        }
-        // If it wasn't the Back key or there's no web page history, bubble up to the default
-        // system behavior (probably exit the activity)
-        if (SystemClock.elapsedRealtime() - lastBackClick > 2000){
-            lastBackClick = SystemClock.elapsedRealtime();
-            Toast.makeText(this, "Click again to close the app",
-            Toast.LENGTH_LONG).show();
-            return true;
+        // Check if the key event was the Back button
+        if ((keyCode == KeyEvent.KEYCODE_BACK)) {
+            // If there's some web history, go back
+            if(mWebView.canGoBack()) {
+                mWebView.goBack();
+                return true;
+            }
+            // Otherwise, if it's been a little while since the Back key was
+            // last pressed, register that it's been pressed and notify the
+            // user that pressing it again will close the app.
+            if (SystemClock.elapsedRealtime() - lastBackClick > 2000){
+                lastBackClick = SystemClock.elapsedRealtime();
+                Toast.makeText(this, "Click again to close the app",
+                Toast.LENGTH_LONG).show();
+                return true;
+            }
         }
-
-        lastBackClick = SystemClock.elapsedRealtime();
```

Comment:
Don't we still want this call? Otherwise, `lastBackClick` will remain set to the value assigned on initialization, and the above if check will probably always be true.

Reply:
Have to admit it's unclear to me why we're using a time check at all here, though.

Reply:
By the time we get to this point, the keyPress either wasn't a back click, or if it was we're handing it to the system to close the app because neither of those two inner `if` blocks got executed.

Reply:
The time check is to implement this convention where pressing the back button twice within some timeframe will close the app.  So basically, if there's no web history to go back to and the back button gets pressed twice within 2 seconds, the keypress gets handed off to the system, which closes the app.

I must admit that the logic implemented a little bit unintuitively, but I didn't want to totally refactor it.

Reply:
It isn't checking if it's within two seconds of `lastBackClick`, it's checking whether it's been more than two seconds: `if (SystemClock.elapsedRealtime() - lastBackClick > 2000)`. 

After your changes, `lastBackClick` is only set at the app startup time and never updated while the app is running. So once it's been > 2000 ms since the app was started, then `SystemClock.elapsedRealtime() - lastBackClick > 2000` check will always return true. If this always returns true, then once `canGoBack` is false, the user will always get the "Click again to close the app" message when they hit the back button, and the super handler will never be invoked.

Reply:
Actually, nevermind, it does get set when the if block is executed. It's just that `lastBackClick` is a misnomer, since we're not actually setting it whenever back is clicked, which was tripping me up. I think this code should be okay as-is. I think this increases the argument for refactoring it, but it's outside the scope of this PR.